File: app/lib/domain/entities/bounding_box.dart
import 'package:equatable/equatable.dart';

class BoundingBox extends Equatable {
  const BoundingBox({
    required this.laMin,
    required this.loMin,
    required this.laMax,
    required this.loMax,
  });

  final double laMin;
  final double loMin;
  final double laMax;
  final double loMax;

  @override
  List<Object?> get props => [laMin, loMin, laMax, loMax];
}

File: api/src/FlightOps.Domain/BoundingBox.cs
namespace FlightOps.Domain;

public sealed record BoundingBox(double LaMin, double LoMin, double LaMax, double LoMax)
{
    public BoundingBox RoundToGrid(double gridDegrees)
    {
        double RoundDown(double value) => Math.Floor(value / gridDegrees) * gridDegrees;
        double RoundUp(double value) => Math.Ceiling(value / gridDegrees) * gridDegrees;

        return new BoundingBox(RoundDown(LaMin), RoundDown(LoMin), RoundUp(LaMax), RoundUp(LoMax));
    }

    public string CacheKey => $"{LaMin:F2}_{LoMin:F2}_{LaMax:F2}_{LoMax:F2}";
}
